packages/pos-subgraph/src/pool.ts

```diff
@@ -337,6 +343,42 @@ export function handleWithdraw(event: Withdraw): void {
     }
 }
 
+function getPreviousWeeklyPerf(
```

Comment:
This is good approach. Instead of looking the previous weekly/monthly value, you calculate on the fly